### ISSUE TITLE
Image sizes

### ISSUE THREAD

**Issue by mattpilott** [nk-crew/lazy-blocks#68]
Could we please get an image size selection and urls for all sizes

thanks

**Comment by dabasti** [comment]
#73 could cover this

**Comment by iannacone** [comment]
hi,
i created an handlebar helper for it:

```
	$handlebars->registerHelper('img', function($img) {
		
		$srcset = wp_get_attachment_image_srcset($img['id']);
		$info = wp_get_attachment_image_src($img['id'], 'large');
		// $width = $info['width'];
		$width = $info[1];
		
		return '<img src="' . $img['url'] . '" alt="' . $img['alt'] . '" class="wp-image-' . $img['id'] . '" srcset="' . $srcset . '" sizes="(max-width: ' . $width . 'px) 100vw, ' . $width . 'px" />';
		
	});
```

**Comment by iannacone** [comment]
if you have to do something more complicated you can push a variable inside the handlebar context:

```
$handlebars->registerHelper('img-metadata', function($img, $id, $options = null) {
	
	$metadata = wp_get_attachment_metadata($img['id']);
	$options['context']->push([$id => $metadata]);
	
});
```

usage:
`{{img-metadata my_image_variable 'my_new_image_variable_name'}}`

**Comment by pixeldesignstudio** [comment]
Pleeeease add this as an easy option!! 
I don't know how to use those handlebars.

**Comment by dabasti** [comment]
> Pleeeease add this as an easy option!!
> I don't know how to use those handlebars.

You can use PHP code as well if you don't want to use handlebars.
https://lazyblocks.com/documentation/blocks-code/php/

**Comment by pixeldesignstudio** [comment]
> > Pleeeease add this as an easy option!!
> > I don't know how to use those handlebars.
> 
> You can use PHP code as well if you don't want to use handlebars.
> https://lazyblocks.com/documentation/blocks-code/php/

Thx, I tried but I think I am too stupid for this and I am not a beginner.

**Comment by rawsta** [comment]
When using PHP, have you tried this?:
```
<?php
$image = get_lzb_meta( 'control_meta_name' );
echo wp_get_attachment_image( $image['id'], 'large');
```
that should give you a bit more control over the sizes.

**Comment by pixeldesignstudio** [comment]
> When using PHP, have you tried this?:
> 
> ```
> <?php
> $image = get_lzb_meta( 'control_meta_name' );
> echo wp_get_attachment_image( $image['id'], 'large');
> ```
> 
> that should give you a bit more control over the sizes.

Yes, I managed to do it by the PHP handle. But switching the complete code to PHP instead of handlebars is a bit of a turndown.

**Comment by pixeldesignstudio** [comment]
> my_image_variable

Can you please, do me the favor and explain it for me please.
I tried a lot, but maybe over-complicating things.

**1. In my function.php**
```
function addLazyblocksHandlebars($handlebars) {
	$handlebars->registerHelper('img-metadata', function($img, $id, $options = null) {

		$metadata = wp_get_attachment_metadata($img['id']);
		$options['context']->push([$id => $metadata]);
		//Usage: {{img-metadata my_image_variable 'my_new_image_variable_name'}}

	});
}
add_filter( 'addLazyblocksHandlebars', 'add_lazyblockshandlebars' );
```

**2. Using the new handlebar:** 
`{{img-metadata my_image_variable 'hero_image'}}`

**Comment by netzgestaltung** [comment]
Hi there,
just in case someone else is searching for that topic:

I made a helper that implements `wp_get_attachment_image`.

Usage:
`{{{ wp_get_attachment_image control_name 'thumbnail' }}}`

```
/**
 * lazy block Handlebars helper
 * 
 * image sizes helper
 * https://github.com/nk-o/lazy-blocks/issues/68
 */
function myPlugin_lazyblock_handlebars_helper($handlebars){
  $handlebars->registerHelper('wp_get_attachment_image', function($image, $size=null){
    if ( isset($image['id']) ) {
      return wp_get_attachment_image($image['id'], $size);
    }	  
  });
}
// lazy block Handlebars helper
add_action('lzb_handlebars_object', 'myPlugin_lazyblock_handlebars_helper');
  
```

**Comment by claytonschase** [comment]
@pixeldesignstudio not sure if you were needing to accomplish the same thing that I was working on but I found a pretty solid solution for selecting the sizes in the frontend block settings.

I created a dropdown control and assigned the default image options and values so the user can choose the appropriate size when adding the block. Then using the below code in the block I just output the user selected size. Seems to work great. Anyway, I hope this helps anyone else trying to achieve the same thing.

`<?php
$size = $attributes['image-size'];

if ( isset( $attributes['image']['id'] ) ) {
    echo wp_get_attachment_image( $attributes['image']['id'], $size );
}
?>`
